src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -8263,6 +8282,34 @@ public void setIPAddressPreference(String iPAddressPreference) {
     public String getIPAddressPreference() {
         return activeConnectionProperties.getProperty(SQLServerDriverStringProperty.IPADDRESS_PREFERENCE.toString());
     }
+
+
+
+    /** Message handler */
+    private transient ISQLServerMessageHandler serverMessageHandler;
+    
+    /**
+     * Set current message handler
+     * 
+     * @param messageHandler
+     * @return The previously installed message handler (null if none)
```

Comment:
why does the setter need to return anything? there is already a getter

Reply:
This is just convenience...   
Meaning: not have to call getter if you want to save the previous installed handler...   
Which may be used if you want to restore previous installed handler at a later point!

For example the java Properties.getProperty("someName") does the same thing!
https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#setProperty-java.lang.String-java.lang.String-